Source/WebSocket/WinHTTP/winhttp_websocket.cpp

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pch.h"
+
+#if HC_WINHTTP_WEBSOCKETS
+#if !HC_NOWEBSOCKETS
+
+#include "../hcwebsocket.h"
+#include "../../HTTP/WinHttp/winhttp_http_task.h"
+#include "../../global/global.h"
+#include "uri.h"
+
+using namespace xbox::httpclient;
+
+
+struct winhttp_websocket_impl : public hc_websocket_impl, public std::enable_shared_from_this<winhttp_websocket_impl>
+{
+public:
+    winhttp_websocket_impl(HCWebsocketHandle hcHandle)
+    {
+        m_hcWebsocketHandle = HCWebSocketDuplicateHandle(hcHandle);
+        // TODO: when to close
```

Comment:
>/ TODO: when to close [](start = 9, length = 21)

I think you don't actually want to duplicate the handle here.  iirc this object is owned by the HC_WEBSOCKET, so duplicated the handle here will cause it to never get cleaned up.

Reply:
the websocket_impl object's lifetime is always the same as the HC_WEBSOCKET

---
In reply to: [248444615](https://github.com/Microsoft/libHttpClient/pull/361#discussion_r248444615) [](ancestors = 248444615)